server/src/main/java/org/opensearch/tasks/TaskManager.java

```diff
@@ -202,6 +209,11 @@ public void cancel(CancellableTask task, String reason, Runnable listener) {
      */
     public Task unregister(Task task) {
         logger.trace("unregister task for id: {}", task.getId());
+
+        if (taskResourceTrackingService.get() != null && task.supportsResourceTracking()) {
+            taskResourceTrackingService.get().stopTracking(task);
+        }
```

Comment:
This is majorly added to not break unit tests. In actual code there'll be no concurrency issue.

As in unit tests there's no stopping for anyone to instantiate TaskManager directly (without going through a common mock class). As taskResourceTracking service is not part of the constructor params and needs to be set separately it becomes a huge task for setting this dependency explicitly for all those tests.

Reply:
Shouldn't assert be better resort in that case?

Reply:
This is added to avoid unit test failures only. Since unit test can initialise task manager and use it in tests without setting this dependency. So there's no single place to set this dependency for all task manager instances in unit tests. Hence, added a null check here